src/arithmetic/canonical.cc

```diff
@@ -481,41 +481,73 @@ class Canonical::Internal : public IRMutator {
     }
     return value;
   }
-  // Detect if a = x * coeff + y, where y \in [0, coeff), x >= 0
-  // return true if such detection is successful
-  // return false if it is not.
+  // Detect if a = q * coeff + r, where r \in [0, coeff), coeff > 0
+  // (in Euclidean division)
+  // returns pair (q, r) if such detection is successful
+  // returns empty vector otherwise.
+  // Assumes that coeff is a constant integer
   std::vector<ComExpr> TryLinearEquation(const ComExpr& a,
                                          const Expr& coeff) {
     Type type = coeff.type();
     int64_t value = GetConstIntValue(coeff);
     if (value < 0) return {};
```

Comment:
May be `<=` instead of `<`? I fear that the compiler might crash when `value == 0` and I'm not sure if there is a test for it.